main.go

```diff
@@ -8,6 +8,11 @@ import (
 	log "github.com/Sirupsen/logrus"
 	"github.com/rancherio/go-machine-service/events"
 	"github.com/rancherio/go-machine-service/handlers"
+	_ "github.com/rancherio/go-machine-service/handlers/providers/amazonec2"
```

Comment:
I understand that they are stateless, but I disagree with your assessment that that is the only reason they should be explicitly initialized in a central place.

The very fact that you have to import and ignore these providers is a code smell. It means you are making an explicit dependency implicit. Your just obfuscating what you're doing and it's not buying you anything because you still need to import the providers here.

If I were looking at this code as developer new to go-machine-service, I would have to do a decent amount of code swimming to understand why you are importing and ignoring these things and that behind the scenes, that is actually wiring up a required dependency.